api/proto/info/info.proto

```diff
@@ -393,6 +394,37 @@ message ZCellularProvider {
   bool roaming = 4;
 }
 
+// Snapshot states
+enum ZSnapshotState {
+  Z_SNAPSHOT_STATE_UNSPECIFIED = 0;
+  // This state is used when a snapshot is in the process of being
+  // created or an error occurred during the first attempt to create it.
+  // (For example, the operation was delayed)
+  Z_SNAPSHOT_STATE_CREATING = 1;
+  // This state is used when the snapshot has been successfully created.
+  Z_SNAPSHOT_STATE_CREATED = 2;
+  // This state is used when the snapshot is pending deletion or
+  // the first deletion attempt was not successful.
+  Z_SNAPSHOT_STATE_DELETING = 3;
+  // This state is used to send information to the controller about a
+  // snapshot that was implicitly deleted after a rollback snapshot
+  // or volume delete command.
+  Z_SNAPSHOT_STATE_IMPLICITLY_DELETED = 4;
+}
+
+// ZInfoSnapshot - Information about snapshot in zfs for zvol
+message ZInfoSnapshot {
+  uint64 creation_time = 1;           // In seconds
+  string uuid = 2;                    // link a command and a response. Is the real name of the snapshot in ZFS
+  string volume_uuid = 3;             // Volume ID of the volume this snapshot belongs to.
+  string display_name = 4;            // Ex: "Tuesday" or creation time (User-friendly name)
+  bool encryption = 5;
+  ZSnapshotState current_state = 6;   // Displays the current state of the snapshot
+  string error_msg = 7;               // Ops error
```

Comment:
We have a message ErrorInfo which makes sense to use here to get timestamp, severity, etc.

Reply:
Corrected